PusherClient/Pusher.cs

```diff
@@ -100,24 +101,14 @@ public Pusher(string applicationKey, PusherOptions options = null)
 
         #region Public Methods
 
+        [MethodImpl(MethodImplOptions.Synchronized)]

```

Comment:
It's basically nicer, and doesn't expose the locking mechanism in a public way.  The proposed method in the PR can create deadlocks :(

Here's an interesting SO question on this very matter. http://stackoverflow.com/questions/6140048/difference-between-manual-locking-and-synchronized-methods :)